simde/x86/sse4.2.h

```diff
@@ -37,6 +37,32 @@ SIMDE__BEGIN_DECLS
 #  define SIMDE_X86_SSE4_2_ENABLE_NATIVE_ALIASES
 #endif
 
+SIMDE__FUNCTION_ATTRIBUTES
+int
+simde_mm_cmpestrs_16_(int la) {
+  const int upper_bound = (128/16) - 1;
+  return la <= upper_bound;
+}
+
+SIMDE__FUNCTION_ATTRIBUTES
+int
+simde_mm_cmpestrs_8_(int la) {
+  const int upper_bound = (128/8) - 1;
+  return la <= upper_bound;
+}
+
+#if defined(SIMDE_X86_SSE4_2_NATIVE)
+#  define simde_mm_cmpestrs(a, la, b, lb, imm8) _mm_cmpestrs(a, la, b, lb, imm8)
+#else
+#  define simde_mm_cmpestrs(a, la, b, lb, imm8) \
+     (((imm8) & SIMDE_SIDD_UWORD_OPS) \
+       ? simde_mm_cmpestrs_16_(la) \
+       : simde_mm_cmpestrs_8_(la))
+#endif
+#if defined(SIMDE_X86_SSE4_2_ENABLE_NATIVE_ALIASES)
```

Comment:
You're right, this is simpler.  So much so, in fact, that I don't think it makes sense to break this into separate functions.  You should only break up code like this into separate functions when it makes things easier to understand, but this just makes things more complicated.  This could be implemented simply as `return la < ((128 / ((imm8 & SIMDE_SIDD_UWORD_OPS) ? 16 : 8)) - 1);`, plus of course the native implementation.  And we can put a `SIMDE_REQUIRE_CONSTANT_RANGE(imm8, 0, 0x7f)` on it to validate the input.